lib/jsdom/living/helpers/http-request.js

```diff
@@ -0,0 +1,261 @@
+"use strict";
+const http = require("http");
+const https = require("https");
+const { Writable } = require("stream");
+const zlib = require("zlib");
+
+const ver = process.version.replace("v", "").split(".");
+const majorNodeVersion = Number.parseInt(ver[0]);
+
+function abortRequest(clientRequest) {
+  // clientRequest.destroy breaks the test suite for versions 10 and 12,
+  // hence the version check
+  if (majorNodeVersion > 13) {
+    clientRequest.destroy();
+  } else {
+    clientRequest.abort();
+  }
+  clientRequest.removeAllListeners();
+  clientRequest.on("error", () => {});
+}
+
+module.exports = class Request extends Writable {
+  constructor(url, options) {
+    super();
+    this._currentURL = url;
+    this._options = options;
+    this.headers = this._options.headers;
```

Comment:
It looks like some of the XHR code accesses `realClient._options.headers`. Which is bad because it's accessing underscore-prefixed properties. Which makes me wonder why we have two copies of this property anyway.

Maybe the better pattern would be to have a public getter, e.g. `get headers() { return this._options.headers; }`?

Although even then there's a _third_ source of truth, right? `this._currentRequest.headers`? Hmm.